### DISCUSSION TITLE
authExchange cancelling request, but still executing it

### DISCUSSION THREAD

**Discussion by hexwit** [urql-graphql/urql#1813] [Q&A]
Hi

I know that subject sound confusing, but here is the situation: 

I have authExchange that implements getAuth and addAuthToOperation methods.
When i make un-authenticated request to the server - request executed (I see logs on the server), 
but browser shows that request has been cancelled.

getAuth returns null, and addAuthToOperation has default implemented like in your example.

Is this by design or I miss something?

**Comment by kitten** [comment]
without code we probably can't really tell what's going on. What's immediately suspicious to me is, did you implement the functions that tell `authExchange` what to consider as an error? Also, you mention cancelled requests, and that sounds more like the requests are being sent but your UI or what's starting the queries isn't listening to the results any longer

**Reply by hexwit** [reply]
you are right! I am doing navigation the login page via location.href in the getAuth method, when it is clear that no auth info exists. What is the preferred way to do a redirect in such case?

**Reply by kitten** [reply]
That's completely fine to do; The thing is, the cancellation is expected behaviour then though possibly. If your queries aren't needed anymore any inflight requests will be cancelled
